recipes/asio-grpc/all/conanfile.py

```diff
@@ -0,0 +1,82 @@
+from conans import ConanFile, CMake, tools
+import os
+
+required_conan_version = ">=1.43.0"
+
+
+class AsioGrpcConan(ConanFile):
+    name = "asio-grpc"
+    description = ("Asynchronous gRPC with Asio/unified executors")
+    homepage = "https://github.com/Tradias/asio-grpc"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    topics = ("cpp", "asynchronous", "grpc", "asio", "asynchronous-programming", "cpp17", "coroutine", "cpp20", "executors")
+    settings = "compiler"
+    no_copy_source = True
+    options = {
+        "backend": ["boost", "asio", "unifex"],
+        "use_boost_container": ["auto", True, False],
+    }
+    default_options = {
+        "backend": "boost",
+        "use_boost_container": "auto",
+    }
+
+    def validate(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            tools.check_min_cppstd(self, "17")
+
+    def configure(self):
+        if self.options.use_boost_container == "auto":
+            self.options.use_boost_container = self.settings.compiler.get_safe("libcxx") == "libc++" or \
+                (self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "9")
+
+    def requirements(self):
+        self.requires("grpc/1.45.2")
+        if self.options.use_boost_container or self.options.backend == "boost":
+            self.requires("boost/1.79.0")
+        if self.options.backend == "asio":
+            self.requires("asio/1.22.1")
+        if self.options.backend == "unifex":
+            self.requires("libunifex/cci.20220430")
+
+    def package_id(self):
+        self.info.header_only()
+        self.info.options.use_boost_container = self.options.use_boost_container
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True)
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses")
+        cmake = CMake(self)
+        cmake.definitions["ASIO_GRPC_USE_BOOST_CONTAINER"] = self.options.use_boost_container
+        cmake.configure()
+        cmake.install()
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib", "cmake", "asio-grpc"), "asio-grpc*")
+
+    def package_info(self):
+        self.cpp_info.builddirs = [os.path.join("lib", "cmake", "asio-grpc")]
+        build_modules = [os.path.join(self.cpp_info.builddirs[0], "AsioGrpcProtobufGenerator.cmake")]
```

Comment:
We typically never bundled installed CMake scripts... if this just purely helper functions?

Reply:
Yes it contains one [helper function](https://github.com/Tradias/asio-grpc/blob/master/cmake/AsioGrpcProtobufGenerator.cmake) for end users and is considered part of [asio-grpc's public API](https://tradias.github.io/asio-grpc/md_doc_cmake_protobuf_generate.html). I have seen that other recipes like protobuf (which provides a similar function) is doing the same thing.